openshift_metrics/openshift_prometheus_metrics.py

```diff
@@ -0,0 +1,62 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+#
+
+import argparse
+import datetime
+import os
+import sys
+
+import openshift as oc
+
+import utils
+
+
+CPU_METRIC = 'kube_pod_init_container_resource_requests_cpu_cores'
+ALLOCATED_CPU_METRIC = 'kube_pod_init_container_resource_limits_cpu_cores'
+ALLOCATED_MEMORY_METRIC = 'kube_pod_init_container_resource_limits_memory_bytes'
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--openshift-url", help="OpenShift Prometheus URL",
+                        default=os.getenv('OPENSHIFT_PROMETHEUS_URL'))
+    parser.add_argument("--report-date", help="report date (ex: 2022-03-14)",
+                        default=(datetime.datetime.today() - datetime.timedelta(days=1)).strftime('%Y-%m-%d'))
+
+    args = parser.parse_args()
+    if not args.openshift_url:
+        sys.exit('Must specify --openshift-url or set OPENSHIFT_PROMETHEUS_URL in your environment')
+    openshift_url = args.openshift_url
+    report_date = args.report_date
+
+    print("Generating report for %s" % report_date)
+
+    token = oc.get_auth_token()
+
+    cpu_metrics = utils.query_metric(openshift_url, token, CPU_METRIC, report_date)
+    allocated_cpu_metrics = utils.query_metric(
+        openshift_url, token, ALLOCATED_CPU_METRIC, report_date)
+    allocated_memory_metrics = utils.query_metric(
+        openshift_url, token, ALLOCATED_MEMORY_METRIC, report_date)
+
+    metrics_dict = {}
+    utils.merge_metrics('cpu', cpu_metrics, metrics_dict)
+    utils.merge_metrics('allocated_cpu', allocated_cpu_metrics, metrics_dict)
+    utils.merge_metrics('allocated_memory', allocated_memory_metrics, metrics_dict)
+    condensed_metrics_dict = utils.condense_metrics(
+        metrics_dict, ['cpu', 'allocated_cpu', 'memory'])
+
+    file_name = "%s.log" % report_date
```

Comment:
Would it make sense to add a `--output` option for output to an explicit filename?

Reply:
Seems like a good idea - added!